PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/localrun/FunctionRunConfiguration.java

```diff
@@ -0,0 +1,176 @@
+import com.intellij.execution.Executor;
+import com.intellij.execution.configurations.ConfigurationFactory;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.configurations.RunProfileState;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.options.SettingsEditor;
+import com.intellij.openapi.project.Project;
+import com.microsoft.intellij.runner.AzureRunConfigurationBase;
+import com.microsoft.intellij.runner.functions.core.FunctionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.nio.file.Paths;
+import java.util.Map;
+
+public class FunctionRunConfiguration extends AzureRunConfigurationBase<FunctionRunModel> {
+
+    private JsonObject appSettingsJsonObject;
+    private FunctionRunModel functionRunModel;
+
+    private Module module;
+
+    protected FunctionRunConfiguration(@NotNull Project project, @NotNull ConfigurationFactory factory, String name) {
+        super(project, factory, name);
+        this.functionRunModel = new FunctionRunModel(project);
+    }
+
+    @Override
+    public FunctionRunModel getModel() {
+        return functionRunModel;
+    }
+
+    @Override
+    public String getTargetName() {
+        return "null";
+    }
+
+    @Override
+    public String getTargetPath() {
+        return "null";
+    }
+
+    @Override
+    public String getSubscriptionId() {
+        return "null";
+    }
+
+    @NotNull
+    @Override
+    public SettingsEditor<? extends RunConfiguration> getConfigurationEditor() {
+        return new FunctionRunSettingEditor(getProject(), this);
+    }
+
+    @Nullable
+    @Override
+    public RunProfileState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment executionEnvironment) throws ExecutionException {
+        return new FunctionRunState(getProject(), this, executor);
+    }
+
+    public Module getModule() {
+        return module == null ? FunctionUtils.getFunctionModuleByFilePath(getProject(), getModuleFilePath()) : module;
+    }
+
+    public JsonObject getAppSettingsJsonObject() {
+        return appSettingsJsonObject;
+    }
+
+    public String getDebugOptions() {
+        return functionRunModel.getDebugOptions();
+    }
+
+    public void setDebugOptions(String debugOptions) {
+        functionRunModel.setDebugOptions(debugOptions);
+    }
+
+    public String getStagingFolder() {
+        return functionRunModel.getStagingFolder();
+    }
+
+    public String getFuncPath() {
+        return functionRunModel.getFuncPath();
+    }
+
+    public String getHostJsonPath() {
+        return functionRunModel.getHostJsonPath();
+    }
+
+    public String getLocalSettingsJsonPath() {
+        final String path = functionRunModel.getLocalSettingsJsonPath();
```

Comment:
functionRunModel.getLocalSettingsJsonPath() is the text input path from user?
Paths.get(getProject().getBasePath(), "local.settings.json") is the default one?

Reply:
Plan to enable user select local settings json later, for the first version, it will use the default value `Paths.get(getProject().getBasePath(), "local.settings.json") `